### PR TITLE
doc/user: add reference docs for AWS PrivateLink connections

### LINKED ISSUE #15920: Reference documentation for AWS PrivateLink connections
*Part of epic https://github.com/MaterializeInc/cloud/issues/4285.*

Add reference documentation for the new AWS PrivateLink connection type (#15797). This should cover:

  * `CREATE CONNECTION ... TO AWS PRIVATELINK`
  * The `AWS PRIVATELINK` options for CSR, Kafka, and Postgres connections

This is in addition to #15342, which is about guide-level documentation for AWS PrivateLink.

### Related work

_No response_

### PR DESCRIPTION
**Pull request by benesch**
And also improve the `CREATE CONNECTION` documentation in preparation of the new AWS PrivateLink connection documentation. Details in commit messages.

Fix #15920.

### Motivation

* Documentation.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: doc/user/content/sql/alter-connection.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Throughout the entire process, the SSH bastion server is configured to permit
+authentication from at least one of the keys that Materialize will authenticate
+with, so Materialize's ability to connect is never interrupted.
+
+You must take care to update the SSH bastion server with the new keys after
+every execution of the `ROTATE KEYS` command. If you rotate keys twice in
+succession without adding the new keys to the bastion server, Materialize will
+be unable to authenticate with the bastion server.
+
+## See also
```
Comment (morsapaes) [inline doc/user/content/sql/alter-connection.md:51]:
```suggestion
## Related pages
```

File: doc/user/content/sql/create-connection.md [hunk 2]
```diff
@@ -8,19 +8,123 @@ menu:
...
-statements.
+A connection describes how to connect and authenticate to an external system you want Materialize to read data from. Once created, a connection is **reusable** across multiple [`CREATE SOURCE`](/sql/create-source), [`CREATE SINK`](/sql/create-sink), and `CREATE CONNECTION` statements.
 
-To use credentials that contain sensitive information (like passwords and SSL keys) in a connection, you must first [create a secret](/sql/create-secret) to securely store them in Materialize's secret management system. Credentials that are generally not sensitive (like usernames and SSL certificates) can be specified as plain `text`, or also stored as secrets.
+To use credentials that contain sensitive information (like passwords and SSL keys) in a connection, you must first [create secrets](/sql/create-secret) to securely store each credential in Materialize's secret management system. Credentials that are generally not sensitive (like usernames and SSL certificates) can be specified as plain `text`, or also stored as secrets.
 
 [//]: # "TODO(morsapaes) Adapt once sinks are wired up to use connections."
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:15]:
```suggestion
```

File: doc/user/content/sql/create-connection.md [hunk 3]
```diff
@@ -8,19 +8,123 @@ menu:
 
 ---
 
-A connection describes how to connect and authenticate to an external system you want Materialize to read data from. Once created, a connection is **reusable** across multiple [`CREATE SOURCE`](/sql/create-source) and [`CREATE SINK`](/sql/create-sink)
-statements.
+A connection describes how to connect and authenticate to an external system you want Materialize to read data from. Once created, a connection is **reusable** across multiple [`CREATE SOURCE`](/sql/create-source), [`CREATE SINK`](/sql/create-sink), and `CREATE CONNECTION` statements.
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:11]: I assume you added `CREATE CONNECTION` to cover CSR, SSH and PrivateLink connections, but this makes it pretty confusing. It should be enough to mention it under each of these connection types.

File: doc/user/content/sql/create-connection.md [hunk 4]
```diff
@@ -8,19 +8,123 @@ menu:
...
+documentation for details.
+
+{{< warning >}}
+Do **not** grant access to the entire Materialize AWS account (`arn:aws:iam::664411391173:root`). Doing so will allow **any** Materialize
+customer to create a connection to your AWS PrivateLink service.
+{{< /warning >}}
+
+### Accepting connection requests
+
+If your AWS PrivateLink service is configured to to require acceptance of
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:72]:
```suggestion
If your AWS PrivateLink service is configured to require acceptance of
```

File: doc/user/content/sql/create-connection.md [hunk 5]
```diff
@@ -8,19 +8,123 @@ menu:
...
+connection requests, you must manually approve the connection request from
+Materialize after executing `CREATE CONNECTION`. See [Accept or reject
+connection requests](https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html#accept-reject-connection-requests) in the
+AWS PrivateLink documentation.
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:76]:
```suggestion
AWS PrivateLink documentation for more details.
```

File: doc/user/content/sql/create-connection.md [hunk 6]
```diff
@@ -8,19 +8,123 @@ menu:
...
+
+### Example
+
+```sql
+CREATE CONNECTION privatelink_svc TO AWS PRIVATELINK (
+    SERVICE NAME 'com.amazonaws.vpce.us-east-1.vpce-svc-0e123abc123198abc',
+    AVAILABILITY ZONES ('use1-az1', 'use1-az4')
+)
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:84]:
```suggestion
);
```

File: doc/user/content/sql/create-connection.md [hunk 7]
```diff
@@ -8,19 +8,123 @@ menu:
...
+```
+
+## Confluent Schema Registry
+
+A Confluent Schema Registry connection establishes a link to a [Confluent Schema
+Registry] server.
+
+You can use Confluent Schema Registry connections in the [`FORMAT`] clause of
+[`CREATE SOURCE`] and [`CREATE SINK`].
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:93]:
```suggestion
You can use Confluent Schema Registry connections in [Kafka connections](#kafka).
```

File: doc/user/content/sql/create-connection.md [hunk 8]
```diff
@@ -8,19 +8,123 @@ menu:
...
+    SSL KEY = SECRET csr_ssl_key,
+    SSL CERTIFICATE = SECRET csr_ssl_crt,
+    USERNAME = 'foo',
+    PASSWORD = SECRET csr_password
+);
+```
+
 ## Kafka
 
+A Kafka connection establishes a link to a [Kafka] cluster.
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:123]: Does any validation happen when `CREATE CONNECTION` is run? As in, is this "describing how to connect", or is this "establishing a link"? If we only really connect on `CREATE SOURCE`, I would phrase it differently here. The current blurb for `CREATE SOURCE` is:

> [CREATE SOURCE](https://materialize.com/docs/sql/create-source/) connects Materialize to an external system you want to read data from, and provides details about how to decode and interpret that data.

File: doc/user/content/sql/create-connection.md [hunk 9]
```diff
@@ -91,50 +195,31 @@ CREATE CONNECTION kafka_connection TO KAFKA (
...
+`SSH TUNNEL`                | object name      |          | The name of an [SSH tunnel connection](#ssh-tunnel) through which network traffic should be routed.
 `SSL CERTIFICATE AUTHORITY` | secret or `text` |          | The absolute path to the certificate authority (CA) certificate in PEM format. Used for both SSL client and server authentication. If unspecified, uses the system's default CA certificates.
 `SSL MODE`                  | `text`           |          | Default: `disable`. Enables SSL connections if set to `require`, `verify_ca`, or `verify_full`.
 `SSL CERTIFICATE`           | secret or `text` |          | Client SSL certificate in PEM format.
 `SSL KEY`                   | secret           |          | Client SSL key in PEM format.
 `USER`                      | `text`           | ✓        | Database username.
+`AWS PRIVATELINK`           | object name      |          | The name of an [AWS PrivateLink connection](#aws-privatelink) through which network traffic should be routed.
 
-##### Example {#postgres-example}
+### Example {#postgres-example}
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:222]:
```suggestion
### Examples {#postgres-example}
```

File: doc/user/content/sql/create-connection.md [hunk 10]
```diff
@@ -149,34 +234,98 @@ CREATE CONNECTION pg_connection TO POSTGRES (
 );
 ```
 
-### SSH tunneling {#postgres-ssh}
```
Comment (morsapaes) [inline doc/user/content/sql/create-connection.md:152]: Not sure about adding it in this way, but I can reorg it later. We'll also need a PrivateLink example for Postgres, right?

**Review by morsapaes — APPROVED** [review]
Thank you, @benesch! Feel free to merge it as is, and I can address my own comments as a follow-up. This also fixes #14879.

**Review by bobbyiliev — APPROVED** [review]

**Pull request merged**